### PR TITLE
Add support for predictions app

### PR DESCRIPTION
**Pull request by gregorjerse**

### CODE REVIEW

File: src/resdk/query.py [hunk 1]
```diff
@@ -423,3 +422,40 @@ def _fetch(self):
...
+
+        missing = collections.defaultdict(list)
+        for value in self._cache:
+            if value._field is None:
+                missing[value.field_id].append(value)
+
+        if missing:
+            # Get corresponding annotation field details in a single query and attach it to
+            # the values.
+            for field in self.resolwe.prediction_field.filter(id__in=missing.keys()):
```
Comment (dblenkus) [inline src/resdk/query.py:458]: Use `.iterate()` to make it more robust.

File: src/resdk/resolwe.py [hunk 2]
```diff
@@ -106,26 +113,30 @@ class Resolwe:
...
     # Map resource class to ResolweQuery name
     resource_query_mapping = {
         AnnotationField: "annotation_field",
         AnnotationValue: "annotation_value",
-        Data: "data",
         Collection: "collection",
-        Sample: "sample",
-        Relation: "relation",
-        Process: "process",
+        Data: "data",
```
Comment (dblenkus) [inline src/resdk/resolwe.py:127]: This reordering should go into a separate commit.

**Review by dblenkus — APPROVED** [review]

**Review by JureZmrzlikar — APPROVED** [review]
Great effort, all looks good. Let's start using it and see what else is needed!

File: src/resdk/resources/predictions.py [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+        self.field_id: Optional[int] = None
+
+        #: sample
+        self.sample_id: Optional[int] = None
+        self._sample: Optional[Sample] = None
+        super().__init__(resolwe, **model_data)
+
+    @property
+    @assert_object_exists
+    def modified(self):
```
[the remark this answers is not in the document]
Comment (gregorjerse) [reply src/resdk/resources/predictions.py:145]: Since it is actually the modification date from the user perspective.

When new prediction / annotation is created (that overrides the old one) it can be seen as a modification of an old value. Not sure what is the `correct`  way of handling this, I can also just leave `created` field and remove `modified`.
